core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -128,41 +133,60 @@ private[spark] class CompressedMapStatus(
  * @param numNonEmptyBlocks the number of non-empty blocks
```

Comment:
You should also update the top-level Scaladoc prose description of this class to discuss the new behavior. The current sentence,

```
  * A [[MapStatus]] implementation that only stores the average size of non-empty blocks,
  * plus a bitmap for tracking which blocks are empty.
```,

is inaccurate after your changes.